core/arch/arm/plat-imx/platform_config.h

```diff
@@ -260,74 +237,50 @@
  */
 
 /* emulated SRAM, at start of secure DDR */
-
 #define CFG_CORE_TZSRAM_EMUL_START	0x4E000000
 
 #define TZSRAM_BASE			CFG_CORE_TZSRAM_EMUL_START
 #define TZSRAM_SIZE			CFG_CORE_TZSRAM_EMUL_SIZE
 
 /* Location of trusted dram */
 
-#define CFG_DDR_TEETZ_RESERVED_START	0x4E100000
-#define CFG_DDR_TEETZ_RESERVED_SIZE	0x01F00000
-
-#define CFG_PUB_RAM_SIZE		(1 * 1024 * 1024)
+#define CFG_TEE_RAM_VA_SIZE             (1 * 1024 * 1024)
 #define CFG_TEE_RAM_PH_SIZE		TZSRAM_SIZE
 
-#define TZDRAM_BASE			(CFG_DDR_TEETZ_RESERVED_START)
-#define TZDRAM_SIZE			(CFG_DDR_TEETZ_RESERVED_SIZE - \
-				CFG_PUB_RAM_SIZE)
+#define CFG_TZDRAM_BASE			0x4E100000
+#define CFG_TZDRAM_SIZE			(30 * 1024 * 1024)
```

Comment:
minor: can move to **conf.mk**  (same for `CFG_TEE_RAM_VA_SIZE` above)

Reply:
This is when `CFG_PAGER=y` which is a bit of a special case. I would have to make `TZSRAM_BASE` configurable, and maybe more, which is more scope than I want to take on at this moment. I don't think the pager configuration on IMX is used in production, because it uses DRAM instead of SRAM, and does not offer any advantage over non-paged. We do want to get the pager working on IMX from SRAM, incorporating the fix for issue #1632.

Reply:
So this change will break the config when one enables `CFG_WITH_PAGER=y`.
Maybe you can **not set** these default defines in **conf.mk** when pager is enabled.

Another way could be that you create and upstream a platform config def script file (I.e **plat-imx/configs/imx<yourtarget>_defconfig.mk**) that defines the "official default platform config" (TZDRAM, SHMEM, ...) and have the optee build command to invoke something like:
```bash
make -C path/to/optee_os CFG_OPTEE_CONFIG=path/to/imx<yourtarget>_defconfig.mk
```
This prevents from breaking current and provides means to clearly define the promoted platform default config. This should not prevent forcing configs from something like
```bash
make CFG_OPTEE_CONFIG=core/arch/arm/plat-imx/imx79_defconfig.mk CFG_TEE_CORE_DEBUG=y CFG_TEE_CORE_EMBED_INTERNAL_TESTS=y
```

.
Another way is that we get the approvals from the related imx maintainers for any changes in plat-imx that could break alternate default configs.

Reply:
@etienne-lms NAK for `CFG_OPTEE_CONFIG`. Platform selection should not need anything other than `PLATFORM=platform[-flavor]`.

Reply:
sorry, my mistake, I forgot to specify the platform/platform_flavor. I should have written 
```
make PLATFORM=imx-<xxx> CFG_OPTEE_CONFIG=core/arch/arm/plat-imx/imx79_defconfig.mk CFG_TEE_CORE_DEBUG=y CFG_TEE_CORE_EMBED_INTERNAL_TESTS=y
```

Reply:
I still don't get the point. What information does `CFG_OPTEE_CONFIG=core/arch/arm/plat-imx/imx79_defconfig.mk` convey that the Makefiles could not figure out by themselves? That you want the default config for the given platform ? Well if it's the default then you should not need to specify anything...?

Reply:
I think optee needs to adopt something like Kconfig. The combination of `conf.mk` and `platform_config.h` is a mess and hard to understand. It would be easier to understand if there was a separate file for each platform flavor (`imx6qsabresd_defconfig.mk`, `imx6qsabresd_pager_defconfig.mk`, `mx6sxsabreauto_defconfig.mk`) and so on. But I don't have time to fundamentally rework optee's build system.

> So this change will break the config when one enables `CFG_WITH_PAGER=y.`

For this change I will make sure the config options are still honored when `CFG_WITH_PAGER=y`.

Reply:
@jordanrh1 as I said, the fact that configuration items may be defined either in `conf.mk` or in `platform_config.h` is purely historical. IMO, `platform_config.h` should ideally be empty.

Regarding KConfig: you're right. This option has been discussed a number of times already mainly as a way to make the `conf.mk` stuff more robust (especially wrt. dependencies). It won't magically solve the `conf.mk` vs. `platform_config.h` issue, but it may be a good opportunity to address it at the same time.

In fact, during Linaro Connect in Hong Kong we decided that we should try to introduce KConfig and we are looking for someone to work on this task. Hopefully someone in Linaro will volunteer, but external contributions would be equally appreciated ;-)

Reply:
I've fixed the `CFG_WITH_PAGER=y` case.